tests/dualtor/test_ipinip.py

```diff
@@ -0,0 +1,144 @@
+
+
+pytestmark = [
+    pytest.mark.topology("t0")
+]
+
+
+@pytest.fixture(scope="module")
+def tor(duthosts, rand_one_dut_hostname):
+    """Select a tor as test target."""
+    return duthosts[rand_one_dut_hostname]
+
+
+@pytest.fixture(scope="function")
+def select_test_interface(tor):
+    """Select a random interface to test."""
+    config_facts = tor.get_running_config_facts()
+    muxcable_table = config_facts["MUX_CABLE"]
+    iface = str(random.choice(muxcable_table.keys()))
+    server_ipv4 = muxcable_table[iface]["server_ipv4"]
+    logging.info("select DUT interface %s to test.", iface)
+    return iface, server_ipv4.split("/")[0]
+
+
+@pytest.fixture(scope="function")
+def build_encapsulated_packet(select_test_interface, ptfadapter, tor, tunnel_traffic_monitor):
+    """Build the encapsulated packet sent from T1 to ToR."""
+    _, server_ipv4 = select_test_interface
+    config_facts = tor.get_running_config_facts()
+    try:
+        peer_ipv4_address = [_["address_ipv4"] for _ in config_facts["PEER_SWITCH"].values()][0]
+    except IndexError:
+        raise ValueError("Failed to get peer ToR address from CONFIG_DB")
+
+    tor_ipv4_address = [_ for _ in config_facts["LOOPBACK_INTERFACE"]["Loopback0"]
+                        if is_ipv4_address(_.split("/")[0])][0]
+    tor_ipv4_address = tor_ipv4_address.split("/")[0]
+
+    inner_dscp = random.choice(range(0, 33))
+    inner_ttl = random.choice(range(3, 65))
+    inner_packet = testutils.simple_ip_packet(
+        ip_src="1.1.1.1",
+        ip_dst=server_ipv4,
+        ip_dscp=inner_dscp,
+        ip_ttl=inner_ttl
+    )[IP]
+    packet = testutils.simple_ipv4ip_packet(
+        eth_dst=tor.facts["router_mac"],
+        eth_src=ptfadapter.dataplane.get_mac(0, 0),
+        ip_src=peer_ipv4_address,
+        ip_dst=tor_ipv4_address,
+        ip_dscp=inner_dscp,
+        ip_ttl=255,
+        inner_frame=inner_packet
+    )
+    logging.info("the encapsulated packet to send:\n%s", tunnel_traffic_monitor._dump_show_str(packet))
+    return packet
+
+
+def get_ptf_server_intf_index(tor, tbinfo, iface):
+    """Get the index of ptf ToR-facing interface on ptf."""
+    mg_facts = tor.get_extended_minigraph_facts(tbinfo)
+    return mg_facts["minigraph_ptf_indices"][iface]
+
+
+def build_expected_packet_to_server(encapsulated_packet):
+    """Build packet expected to be received by server from the tunnel packet."""
+    inner_packet = encapsulated_packet[IP].payload[IP].copy()
+    # use dummy mac address that will be ignored in mask
+    inner_packet = Ether(src="aa:bb:cc:dd:ee:ff", dst="aa:bb:cc:dd:ee:ff") / inner_packet
+    exp_pkt = mask.Mask(inner_packet)
+    exp_pkt.set_do_not_care_scapy(Ether, "dst")
+    exp_pkt.set_do_not_care_scapy(Ether, "src")
+    exp_pkt.set_do_not_care_scapy(IP, "tos")
+    exp_pkt.set_do_not_care_scapy(IP, "ttl")
+    exp_pkt.set_do_not_care_scapy(IP, "chksum")
+    return exp_pkt
+
+
+def test_dualtor_ipinip_active_tor(
```

Comment:
A minor suggestion here.
We have a naming rule according to test scenario.
https://github.com/Azure/sonic-mgmt/blob/b3a938f043d3e59b2f771d17171c82216046caad/docs/testplan/dual_tor/dual_tor_test_plan_action_items.md#test-scenarios

Reply:
renamed, done.